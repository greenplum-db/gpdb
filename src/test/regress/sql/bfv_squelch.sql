--
-- https://github.com/greenplum-db/gpdb/issues/11673
--

EXPLAIN ANALYSE SELECT nspname, nspacl, privilege_type, rolname FROM pg_namespace
JOIN lateral  
(SELECT * FROM aclexplode(nspacl) x 
JOIN pg_authid  ON x.grantee = pg_authid.oid) 
z ON true;

SET from_collapse_limit = 1;
SET join_collapse_limit = 1;

EXPLAIN ANALYSE SELECT nspname, nspacl, privilege_type, rolname FROM pg_namespace
JOIN lateral  
(SELECT * FROM aclexplode(nspacl) x 
JOIN pg_authid  ON x.grantee = pg_authid.oid) 
z ON true;