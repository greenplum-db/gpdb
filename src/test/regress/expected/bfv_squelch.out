--
-- https://github.com/greenplum-db/gpdb/issues/11673
--
EXPLAIN ANALYSE SELECT nspname, nspacl, privilege_type, rolname FROM pg_namespace
JOIN lateral  
(SELECT * FROM aclexplode(nspacl) x 
JOIN pg_authid  ON x.grantee = pg_authid.oid) 
z ON true;
QUERY PLAN
___________
{
  'child' => [
    {
      'child' => [
        {
          'id' => 3,
          'parent' => 2,
          'short' => 'Seq Scan on pg_namespace'
        },
        {
          'id' => 4,
          'parent' => 2,
          'short' => 'Function Scan on aclexplode x'
        }
      ],
      'id' => 2,
      'parent' => 1,
      'short' => 'Nested Loop'
    },
    {
      'child' => [
        {
          'id' => 6,
          'parent' => 5,
          'short' => 'Seq Scan on pg_authid'
        }
      ],
      'id' => 5,
      'parent' => 1,
      'short' => 'Hash'
    }
  ],
  'id' => 1,
  'short' => 'Hash Join'
}
GP_IGNORE:(14 rows)

SET from_collapse_limit = 1;
SET join_collapse_limit = 1;
EXPLAIN ANALYSE SELECT nspname, nspacl, privilege_type, rolname FROM pg_namespace
JOIN lateral  
(SELECT * FROM aclexplode(nspacl) x 
JOIN pg_authid  ON x.grantee = pg_authid.oid) 
z ON true;
QUERY PLAN
___________
{
  'child' => [
    {
      'id' => 2,
      'parent' => 1,
      'short' => 'Seq Scan on pg_namespace'
    },
    {
      'child' => [
        {
          'id' => 4,
          'parent' => 3,
          'short' => 'Seq Scan on pg_authid'
        },
        {
          'child' => [
            {
              'id' => 6,
              'parent' => 5,
              'short' => 'Function Scan on aclexplode x'
            }
          ],
          'id' => 5,
          'parent' => 3,
          'short' => 'Hash'
        }
      ],
      'id' => 3,
      'parent' => 1,
      'short' => 'Hash Join'
    }
  ],
  'id' => 1,
  'short' => 'Nested Loop'
}
GP_IGNORE:(14 rows)

